### PR TITLE
Write MULTPV to INIT-file

### PR DESCRIPTION
**Pull request by blattms**
This output was missing from the init file previously. Now MULTPV will be output if it is specified in the GRID or EDIT section. 

Support for MULTPV in the SCHEDULE section will be added in an upcoming PR. I need to sort out the parallel handling of that.

This is is #4015 again

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this ignore_extra please

**Comment by blattms** [comment]
jenkins build this ignore_extra please

**Comment by blattms** [comment]
Cool, finally this works with all existing tests, see [jenkins #6985](https://ci.opm-project.org/job/opm-common-PR-builder/6985/).

The two difference to the previous attempt are:
- I had to move a few methods to delay instantiation to after the functions are specialised
- In `apply_multipliers` (used in the EDIT section) we first check whether PORV is already initialized. Only if that is the case then MULTPV in the EDIT section will also be applied to PORV. This mimics the currenty behaviour (including issues pointed out in #4024)

**Comment by lisajulia** [comment]
I would like to include this into the release candidate 2, so I'll wait until this is merged to master until I create the new release candidate, ok?

**Comment by bska** [comment]
jenkins build this ignore_extra please

File: opm/input/eclipse/EclipseState/Grid/FieldProps.cpp
Comment (bska) [inline opm/input/eclipse/EclipseState/Grid/FieldProps.cpp:782]: Would you mind adding a comment here for why `hasPorvBefore` is the same as `PORV` **not** being in `double_data`, please?  Also, is there a reason for not marking `hasPorvBefore` as `const`?

File: opm/input/eclipse/EclipseState/Grid/FieldProps.cpp
Comment (bska) [inline opm/input/eclipse/EclipseState/Grid/FieldProps.cpp:801]: I'll mention once more that this code would be simpler if you did not explicitly specify the `Scalar` type and instead passed `std::multiplies<>()` (literally) as the final argument to `std::transform()`.  Using the defaulted template parameter (formally `void`) means we get template argument deduction on the parameters passed to the function call operator of `std::multiplies<>` and that is usually what we want.

**Review by bska — COMMENTED** [review]
Looks good for the most part.  We have to be a little bit careful about what we do to the reference solutions in the time leading up to the release, but if the release manager is convinced that this should **also** go into the release then we're okay to update the reference solutions too.

**Comment by blattms** [comment]
I still would like to rerun my local tests for this. Sorry that this takes so long. Somehow things keep popping up...

**Comment by blattms** [comment]
jenkins build this ignore_extra please

**Comment by blattms** [comment]
I checked my testcase now, too.
Review comments addressed, just not sure whether I got what the requested comment should be about.

**Comment by blattms** [comment]
jenkins build this ignore_extra please

**Review by bska — APPROVED** [review]
Thank you for the updates.  This looks good to me now.  Unless you have other changes incoming here I suggest we update the reference solutions and merge this work into the master branch.

**Comment by blattms** [comment]
jenkins build this with update_data please

**Comment by bska** [comment]
The new reference solutions have been installed on the CI system.  Merging this now to activate the INIT file support.

**Pull request merged**
